CHANGES.rst

```diff
@@ -457,6 +457,8 @@ Bug fixes
 
 - ``astropy.table``
 
+  - Fix an issue with setting fill value when column dtype is changed. [#4088]
+
```

Comment:
D'oh, this went in the v1.1 section instead of the v1.0.5 section.

(I'm actually doing some work toward eventually getting the new changelog format up and running so this will never have to happen again...)